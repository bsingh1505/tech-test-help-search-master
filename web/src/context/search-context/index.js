export * from "./search-result-context";
