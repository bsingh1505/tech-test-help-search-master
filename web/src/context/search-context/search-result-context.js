import React, { useCallback, useState } from "react";
import { ITEMS_PER_PAGE } from "../../utils/constants";
import { fetchSearchResultsFromApi } from "../../utils/http";

export const SearchResultContext = React.createContext();
export const SearchResultContextProvider = ({ children }) => {
  const [searchResults, setSearchResults] = useState([]);
  const [totalCount, setTotalCount] = useState(0);
  const [apiError, setApiError] = useState();

  const getSlugValue = (slugs) => {
    return slugs[slugs.length - 1];
  };
  const formatData = (list) => {
    return list.map((item) => ({
      ...item,
      slug: getSlugValue(item.url.split("/")),
    }));
  };

  const fetchSearchResults = useCallback(async (searchFor, page = 0, limit = ITEMS_PER_PAGE) => {
    fetchSearchResultsFromApi(searchFor, page, limit)
      .then((response) => {
        setSearchResults(formatData(response.payload.results));
        setTotalCount(response.payload.total_count);
      })
      .catch((error) => {
        setApiError(true);
      });
  }, []);

  return (
    <SearchResultContext.Provider
      value={{
        fetchSearchResults,
        apiError,
        searchResults,
        totalCount,
      }}
    >
      {children}
    </SearchResultContext.Provider>
  );
};
