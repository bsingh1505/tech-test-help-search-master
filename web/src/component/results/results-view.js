import { useContext } from "react";
import { CardView } from "../card/card";
import { PaginationView } from "../pagination/pagination";
import { SearchResultContext } from "../../context/search-context/search-result-context";

export const SearchResultsView = ({ currentPage, setCurrentPage }) => {
  const { searchResults, totalCount } = useContext(SearchResultContext);
  return (
    <>
      <div>
        {searchResults.map((item) => (
          <CardView data={item} />
        ))}
        <PaginationView totalItems={totalCount} setCurrentPage={setCurrentPage} currentPage={currentPage} />
      </div>
    </>
  );
};
