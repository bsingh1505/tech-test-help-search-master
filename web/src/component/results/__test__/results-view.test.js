import React from "react";
import { shallow } from "enzyme";
import { SearchResultsView } from "../results-view";
import { SearchResultContext } from "../../../context/search-context/search-result-context";

const mockSearchResultContext = {
  fetchSearchResults: jest.fn(),
  searchResults: [],
};

const setupTest = () => {
  const wrapper = shallow(
    <SearchResultContext.Provider value={mockSearchResultContext}>
      <SearchResultsView />
    </SearchResultContext.Provider>
  );
  return {
    wrapper,
  };
};

describe("components/search-box/search-box-view", () => {
  test("should render", () => {
    const { wrapper } = setupTest();
    expect(wrapper.debug).toMatchSnapshot();
  });
});
