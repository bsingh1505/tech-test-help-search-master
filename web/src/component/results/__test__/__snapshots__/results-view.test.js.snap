// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/search-box/search-box-view should render 1`] = `[Function]`;
